src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -113,15 +113,18 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         }
     }
 
+    public getNotebookControllers(): VSCodeNotebookController[] {
```

Comment:
Seems like this should be async and await the load? Just to make sure. Otherwise there's an implicit dependency between the two.

Reply:
@rchiodo Yeah, I'm just playing around a bit in draft now. So right now this function is only used for tests and as I see it, it would always reflect the current state of registered controllers. I don't actually want it to trigger the load, it's just the current state of what is registered for tests.

So you could do in a test:
1. getNotebookControllers check that it's 0
2. open a document
3. Wait for selection
4. getNotebookControllers check that it's the expected number

Reply:
Hmm, still not happy with the public part of it then. I wish typescript had InternalsVisibleTo. Can you put a comment saying this is only for tests?

Reply:
@rchiodo We've talked over this some for other test only interfaces. I think that I've used a specific test only interface in other places. Let me look that up. Might be better here along with a comment. Will update.

Reply:
Ahh, now I remember what I did. For WebViewHost I created a decorator that enforces test only execution. That + a test interface should work here.

Reply:
Added the decorator on the function + comment on the interface.